pkg/apis/batch/validation/validation_test.go

```diff
@@ -229,21 +230,6 @@ func TestValidateJob(t *testing.T) {
 				},
 			},
 		},
-		"valid pod replacement with failed": {
```

Comment:
why remove this test?

Reply:
I don't really know but this PR should not include these changes once the API ones are merged in.

Reply:
The API PR adds this test https://github.com/kubernetes/kubernetes/pull/119301/files#diff-929cc06aae2beac0c15d0227ce58cc35342b624c896922a76ac7ddc0dbaad994R232

but then the second commit in this PR removes it

Reply:
Added back.